bin/run-samples

```diff
@@ -0,0 +1,37 @@
+#!/usr/bin/env ruby
+
+require 'open3'
+require 'thread'
+
+$signals = Queue.new
```

Comment:
huh, globals? Is there a reason we'd use a global here and not an instance variable?